### PR TITLE
Update gateway max resv size to 16MB

### LINKED ISSUE #5752: End point /eth/v1alpha1/beacon/state broken with max size
# 🐞 Bug Report

### Description
The debug RPC endpoint `/eth/v1alpha1/beacon/state` is currently broken due to a beacon state object exceeds default client dialing max send message size at 4MB.

`
~ $ curl -s -X GET "localhost:6000/eth/v1alpha1/beacon/state?slot=819" -H "accept: application/json"
`

`
{"error":"grpc: received message larger than max (5154533 vs. 4194304)","code":8,"message":"grpc: received message larger than max (5154533 vs. 4194304)","details":[]}⏎
`

### PR DESCRIPTION
**Pull request by terencechain**
This fixes #5752 

Change list:
 * Move `grpc-max-msg-size` flag to `shared` folder and to be shared between validator and beacon node
 * Allow `grpc-max-msg-size` flag to be used to initialize gateway

### CODE REVIEW

File: beacon-chain/gateway/gateway.go [hunk 1]
```diff
@@ -154,8 +154,20 @@ func dial(ctx context.Context, network, addr string) (*grpc.ClientConn, error) {
...
 // "addr" must be a valid TCP address with a port number.
-func dialTCP(ctx context.Context, addr string) (*grpc.ClientConn, error) {
-	return grpc.DialContext(ctx, addr, grpc.WithInsecure())
+func (g *Gateway) dialTCP(ctx context.Context, addr string) (*grpc.ClientConn, error) {
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+
+	if g.enableDebugRPCEndpoints {
+		// Increase Max call size from 4MB to 16MB to account for end point to retrieve beacon state,
+		// this should be large enough to cover state size up to 300k validators.
+		opts = append(opts, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(1<<24)))
```
Comment (prestonvanloon) [inline beacon-chain/gateway/gateway.go:163]: Please use a flag for this like we do with the validator.

https://github.com/prysmaticlabs/prysm/blob/546196a6fa8174d68f5c92071ff4bc6edd1ce3d0/validator/flags/flags.go#L37-L40

**Comment by prestonvanloon** [comment]
> Update gateway client's max resv size to 16GB and gated behind enableDebugRPCEndpoints flag

Please update description. Is it MB or GB?

**Comment by terencechain** [comment]
ready again @prestonvanloon

File: beacon-chain/gateway/server/main.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ var (
 	debug                   = flag.Bool("debug", false, "Enable debug logging")
 	allowedOrigins          = flag.String("corsdomain", "", "A comma separated list of CORS domains to allow")
 	enableDebugRPCEndpoints = flag.Bool("enable-debug-rpc-endpoints", false, "Enable debug rpc endpoints such as /eth/v1alpha1/beacon/state")
+	grpcMaxMsgSize          = flag.Int("grpc-max-msg-size", 1<<22, "Integer to define max recieve message call size (default: 4194304 (for 40MB))")
```
Comment (prestonvanloon) [inline beacon-chain/gateway/server/main.go:24]: This is going to print weird since flag --help will print the default value

Comment (prestonvanloon) [inline beacon-chain/gateway/server/main.go:24]:
```suggestion
	grpcMaxMsgSize          = flag.Int("grpc-max-msg-size", 1<<22, "Integer to define max recieve message call size")
```

Comment (terencechain) [reply beacon-chain/gateway/server/main.go]: got it, thanks!

**Pull request merged**